gcc/rust/ast/rust-ast.h

```diff
@@ -1523,6 +1528,9 @@ struct MacroInvocData
   SimplePath &get_path () { return path; }
   const SimplePath &get_path () const { return path; }
 
+  void set_expander (MacroExpander *new_expander) { expander = new_expander; }
+  MacroExpander *get_expander () { return expander; }
```

Comment:
This can produces segfaults if not handled properly. I would suggest changing the `expander` field from
```c++
MacroExpander *expander = nullptr;
```
to
```c++
Optional<MacroExpander&> expander;
```

This is basically the same ergonomis as what you would have with a raw pointer, but forcing you to make sure its content is valid and non-null. The class is defined in `util/rust-optional.h`

Reply:
That's a bit problematic.

You see, `Optional<T>` requires `T` to be a complete type. We can't have that since `rust-macro-expand.h` contains `MacroExpander` but also includes `rust-ast.h` itself!

Reply:
Ah, good catch. Then I would simply add a `rust_assert (expander)` before the return statement to avoid segfaults